src/registry/v1/data/common/labels/mod.rs

```diff
@@ -0,0 +1,146 @@
+#[cfg(feature = "nom")]
+mod parser;
+
+#[cfg(feature = "nom")]
+pub use parser::*;
+
+use futures::future::join;
+use nom::into;
+#[cfg(feature = "nom")]
+use std::convert::TryFrom;
+use std::fmt;
+
+#[derive(Default, Debug)]
+pub struct LabelSelector(pub Vec<Operation>);
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Operation {
+    Eq(String, String),
+    NotEq(String, String),
+    In(String, Vec<String>),
+    NotIn(String, Vec<String>),
+    Exists(String),
+    NotExists(String),
+}
+
+#[cfg(feature = "nom")]
+impl TryFrom<&str> for LabelSelector {
+    type Error = parser::ParserError;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        Ok(LabelSelector(parser::parse_from(value)?))
+    }
+}
+
+#[cfg(feature = "nom")]
+impl TryFrom<String> for LabelSelector {
+    type Error = parser::ParserError;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        Ok(LabelSelector(parser::parse_from(&value)?))
+    }
+}
+
+impl fmt::Display for Operation {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let (left, operation, right) = match self {
+            Operation::Eq(l, r) => (l.clone(), "=".to_string(), r.clone()),
+            Operation::NotEq(l, r) => (l.clone(), "!=".to_string(), r.clone()),
+            Operation::In(l, r) => (
+                l.clone(),
+                " in ".to_string(),
+                ["(", r.join(", ").as_str(), ")"].concat(),
+            ),
+            Operation::NotIn(l, r) => (
+                l.clone(),
+                " notin ".to_string(),
+                ["(", r.join(", ").as_str(), ")"].concat(),
+            ),
+            Operation::Exists(l) => (l.clone(), "".to_string(), "".to_string()),
+            Operation::NotExists(l) => ("!".to_string(), l.clone(), "".to_string()),
+        };
+
+        write!(f, "{}{}{}", left, operation, right)
+    }
+}
+
+impl From<Operation> for LabelSelector {
+    fn from(op: Operation) -> Self {
+        LabelSelector(vec![op])
+    }
+}
+
+impl LabelSelector {
+    /// Add another operation to a label selector.
+    ///
+    pub fn add(mut self, op: Operation) -> Self {
```

Comment:
Not sure if this work, but (checking the check error), couldn't this implement the "add" trait, so that one could write:

```rust
let mut selector = LabelSelector::new();
selector += Operation::eq("foo", "bar");
```

Reply:
`.add` is more intuitive to me tbh